test/unit/Hbar.js

```diff
@@ -18,4 +18,84 @@ describe("Hbar", function () {
 
         expect(tinybar).to.eql(hbar.toTinybars().toString());
     });
+
```

Comment:
Can you add this:
```javascript
    it("[from|to]String() should be reversable", function () {
        expect(Hbar.fromString("1").toTinybars().toString()).to.be.equal("100000000");
        expect(Hbar.fromString("1.151").toTinybars().toString()).to.be.equal("115100000");
        expect(Hbar.fromString("1 tℏ").toTinybars().toString()).to.be.equal("1");
        expect(Hbar.fromString("1 μℏ").toTinybars().toString()).to.be.equal("100");
        expect(Hbar.fromString("1 mℏ").toTinybars().toString()).to.be.equal("100000");
        expect(Hbar.fromString("1 ℏ").toTinybars().toString()).to.be.equal("100000000");
        expect(Hbar.fromString("1 kℏ").toTinybars().toString()).to.be.equal("100000000000");
        expect(Hbar.fromString("1 Mℏ").toTinybars().toString()).to.be.equal("100000000000000");
        expect(Hbar.fromString("1 Gℏ").toTinybars().toString()).to.be.equal("100000000000000000");
        expect(Hbar.fromString("1.151 mℏ").toTinybars().toString()).to.be.equal("115100");
        expect(Hbar.fromString("1.151 ℏ").toTinybars().toString()).to.be.equal("115100000");
        expect(Hbar.fromString("1.151 kℏ").toTinybars().toString()).to.be.equal("115100000000");
        expect(Hbar.fromString("1.151 Mℏ").toTinybars().toString()).to.be.equal("115100000000000");
        expect(Hbar.fromString("1.151 Gℏ").toTinybars().toString()).to.be.equal("115100000000000000");
        expect(Hbar.fromString("+1").toTinybars().toString()).to.be.equal("100000000");
        expect(Hbar.fromString("+1.151").toTinybars().toString()).to.be.equal("115100000");
        expect(Hbar.fromString("+1 Gℏ").toTinybars().toString()).to.be.equal("100000000000000000");
        expect(Hbar.fromString("+1.151 Gℏ").toTinybars().toString()).to.be.equal("115100000000000000");
        expect(Hbar.fromString("-1").toTinybars().toString()).to.be.equal("-100000000");
        expect(Hbar.fromString("-1.151").toTinybars().toString()).to.be.equal("-115100000");
        expect(Hbar.fromString("-1 Gℏ").toTinybars().toString()).to.be.equal("-100000000000000000");
        expect(Hbar.fromString("-1.151 Gℏ").toTinybars().toString()).to.be.equal("-115100000000000000");

        let err = false;

        try {
            Hbar.fromString("1.151 tℏ");
        } catch (error) {
            err = error.toString().includes("Hbar in tinybars contains decimals");
        }

        if (!err) {
            throw new Error("`Hbar.fromString()` did not error");
        }

        try {
            Hbar.fromString("1.151 μℏ");
        } catch (error) {
            err = error.toString().includes("Hbar in tinybars contains decimals");
        }

        if (!err) {
            throw new Error("`Hbar.fromString()` did not error");
        }

        try {
            Hbar.fromString("1.151.")
        } catch (error) {
            err = error.toString().includes("invalid string provided");
        }

        if (!err) {
            throw new Error("`Hbar.fromString()` did not error");
        }

        try {
            Hbar.fromString("1.151 uℏ")
        } catch (error) {
            err = error.toString().includes("invalid string provided");
        }

        if (!err) {
            throw new Error("`Hbar.fromString()` did not error");
        }
    });
```

Reply:
Not exactly related to `Hbar.[from|to]String()`, but can you add
```
        if (!this._valueInTinybar.isInteger()) {
            throw new Error("Hbar in tinybars contains decimals");
        }
``` to the end of the `Hbar` constructor.
This is part of the test I added above.